flink-elastic-catalog/src/main/java/com/getindata/flink/connector/jdbc/catalog/ElasticCatalog.java

```diff
@@ -139,12 +123,47 @@ private Map<String, ScanPartitionProperties> extractScanTablePartitionProperties
         return scanPartitionProperties;
     }
 
+    private Map<String, TimeAttributeProperties> extractTimeAttributeProperties(Map<String, String> properties) {
+        Map<String, TimeAttributeProperties> timeAttributeProperties = new HashMap<>();
+
+        for (Map.Entry<String, String> entry : properties.entrySet()) {
+            String key = entry.getKey();
+            if (!key.startsWith("properties.timeattribute.") || !(key.endsWith(".watermark.column") || key.endsWith(".watermark.delay") || key.endsWith(".proctime.column"))) {
+                continue;
+            }
+            String tableName = key.replace("properties.timeattribute.", "").
+                    replace(".watermark.column", "").
+                    replace(".watermark.delay", "").
+                    replace(".proctime.column", "");
+            boolean timeAttributePropertiesForTableFound = timeAttributeProperties.containsKey(tableName);
+            TimeAttributeProperties timeAttributePropertiesForTable = timeAttributePropertiesForTableFound ? timeAttributeProperties.get(tableName) : new TimeAttributeProperties();
+
+            if (entry.getKey().endsWith(".watermark.column")) {
+                if (timeAttributePropertiesForTable.getWatermarkColumn() == null) {
+                    timeAttributePropertiesForTable.setWatermarkColumn(entry.getValue());
+                }
```

Comment:
what if != null? x3

Reply:
If all 3 timeAttribute properties are specified then `checkTimeAttributeProperties` method should raise an error saying that `Either proctime or watermark properties should be specified for a table` there is a unit test `testFailGetTableTimeAttributesProctimeAndWatermarkProvided` covering this case.